osutil/disks/disks.go

```diff
@@ -158,19 +185,97 @@ func diskFromMountPointImpl(mountpoint string, opts *Options) (*disk, error) {
 		return nil, fmt.Errorf("cannot find mountpoint %q", mountpoint)
 	}
 
-	// TODO:UC20: if the mountpoint is of a decrypted mapper device, then we
-	//            need to trace back from the decrypted mapper device through
-	//            luks to find the real encrypted partition underneath the
-	//            decrypted one and thus the disk device for that partition
-
 	// now we have the partition for this mountpoint, we need to tie that back
 	// to a disk with a major minor, so query udev with the mount source path
 	// of the mountpoint for properties
-	props, err := udevProperties(partMountPointSource)
+	props, err := udevProperties(mountpointSrc)
 	if err != nil && props == nil {
 		// only fail here if props is nil, if it's available we validate it
 		// below
-		return nil, fmt.Errorf("cannot find disk for partition %s: %v", partMountPointSource, err)
+		return nil, fmt.Errorf("cannot find disk for partition %s: %v", mountpointSrc, err)
+	}
+
+	if opts != nil && opts.IsDecryptedDevice {
+		// verify that the mount point is indeed a mapper device, it should:
+		// 1. have DEVTYPE == disk from udev
+		// 2. have dm files in the sysfs entry for the maj:min of the device
+		if props["DEVTYPE"] != "disk" {
+			// not a decrypted device
+			return nil, fmt.Errorf("mountpoint source %s is not a decrypted device", mountpointSrc)
```

Comment:
we seem to produce the same error for different conditions, maybe for debugging we should add more info?

"mountpoint ... decrypted device,  devtype is not disk"